client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java

```diff
@@ -97,26 +98,28 @@ public DfsPartitionReader(
 
     if (endMapIndex != Integer.MAX_VALUE) {
       long fetchTimeoutMs = conf.clientFetchTimeoutMs();
-      try {
-        client = clientFactory.createClient(location.getHost(), location.getFetchPort());
-        TransportMessage openStream =
-            new TransportMessage(
-                MessageType.OPEN_STREAM,
-                PbOpenStream.newBuilder()
-                    .setShuffleKey(shuffleKey)
-                    .setFileName(location.getFileName())
-                    .setStartIndex(startMapIndex)
-                    .setEndIndex(endMapIndex)
-                    .build()
-                    .toByteArray());
-        ByteBuffer response = client.sendRpcSync(openStream.toByteBuffer(), fetchTimeoutMs);
-        streamHandler = TransportMessage.fromByteBuffer(response).getParsedPayload();
-        // Parse this message to ensure sort is done.
-      } catch (IOException | InterruptedException e) {
-        throw new IOException(
-            "read shuffle file from DFS failed, filePath: "
-                + location.getStorageInfo().getFilePath(),
-            e);
+      if (pbStreamHandler == null) {
+        try {
+          client = clientFactory.createClient(location.getHost(), location.getFetchPort());
+          TransportMessage openStream =
+              new TransportMessage(
+                  MessageType.OPEN_STREAM,
+                  PbOpenStream.newBuilder()
+                      .setShuffleKey(shuffleKey)
+                      .setFileName(location.getFileName())
+                      .setStartIndex(startMapIndex)
+                      .setEndIndex(endMapIndex)
+                      .build()
+                      .toByteArray());
+          ByteBuffer response = client.sendRpcSync(openStream.toByteBuffer(), fetchTimeoutMs);
+          streamHandler = TransportMessage.fromByteBuffer(response).getParsedPayload();
+          // Parse this message to ensure sort is done.
+        } catch (IOException | InterruptedException e) {
+          throw new IOException(
+              "read shuffle file from DFS failed, filePath: "
+                  + location.getStorageInfo().getFilePath(),
+              e);
+        }
```

Comment:
lost else branch?

Reply:
For `DfsPartitionReader`, if we have already got a valid `pbStreamHandler` from `BatchOpenStream` request, it means partition file has been sorted and ready to be fetched. We don't need to perform any additional operations.

Reply:
closeStream will use `streamHandler.getStreamId()`

Reply:
`closeStream ` will use `streamHandler.getStreamId()` only when `client` is initialized and send request to worker to open stream. 

https://github.com/apache/celeborn/blob/299a0483d47e4f7fce86c8d914b5f6f8eb26ea3e/client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java#L282-L294

You can also refer to the scenario when `DfsPartitionReader` read a non-sorted file and there is also no initialization of  `streamHandler`

https://github.com/apache/celeborn/blob/299a0483d47e4f7fce86c8d914b5f6f8eb26ea3e/client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java#L129-L132

Reply:
You are correct, but the closeStream method should create a client to send a request to the worker to end the stream. This allows the worker to safely delete the original file. It's important to note that the stream must also be opened before reading. Therefore, it's incorrect to create the client only during the map range read, so the check for endMapIndex != Integer.MAX_VALUE should be removed in DfsPartitionReader. For more details, please refer to https://github.com/apache/celeborn/pull/1932

Reply:
Fixed. Please review it again.